homeassistant/components/samsungtv/bridge.py

```diff
@@ -232,7 +241,18 @@ def try_connect(self):
             if result:
                 return result
 
-        return RESULT_CANNOT_CONNECT
+        return RESULT_NOT_SUCCESSFUL
+
+    def device_info(self):
+        """Try to gather infos of this TV."""
+        remote = self._get_remote()
+        if remote:
+            try:
+                return remote.rest_device_info()
+            except HttpApiError:
+                # unable to get, ignore
+                pass
+        return None
```

Comment:
This can be written as:

```suggestion
        if not remote:
            return None
        with contextlib.suppress(HttpApiError)
            return remote.rest_device_info()
```

Reply:
Code updated